client/src/app/components/schema-defined-fields/SchemaDefinedFields.tsx

```diff
@@ -13,32 +13,31 @@ export { Language } from "@patternfly/react-code-editor";
 
 type ControlledEditor = {
   focus: () => void;
-  getPosition: () => object; // Add getPosition if you need to read cursor position
+  getPosition: () => object;
   setPosition: (position: object) => void;
-  getValue: () => string; // Important for getting the current editor content
+  getValue: () => string;
 };
 
 export interface ISchemaDefinedFieldProps {
-  // If you wanted to make it dynamic to load other schemas:
   jsonDocument: object | string;
-  jsonSchema: object | string;
+  jsonSchema?: object | string;
   onSchemaSaved?: (newSchemaContent: object) => void;
 }
 
 export const SchemaDefinedField = ({
-  onSchemaSaved = () => {},
+  onSchemaSaved,
   jsonDocument,
   jsonSchema,
 }: ISchemaDefinedFieldProps) => {
   const editorRef = React.useRef<ControlledEditor>();
 
   console.log("SchemaDefinedField rendered with jsonDocument:", jsonDocument);
+
   const initialCode = React.useMemo(
     () => JSON.stringify(jsonDocument, null, 2),
     [jsonDocument]
   );
-  const [currentCode, setCurrentCode] = React.useState(initialCode); // State to hold editable content
-
+  const [currentCode, setCurrentCode] = React.useState(initialCode);
```

Comment:
The `jsonDocument` param is speced to be an object or a string.  If it is a string, we don't want to stringify it.  Add a typeof test to handle the variations properly.